C/services/south-plugin-interfaces/python/python_plugin_interface.cpp

```diff
@@ -145,6 +147,238 @@ void *PluginInterfaceInit(const char *pluginName, const char * pluginPathName)
     return pModule;
 }
 
+/**
+ * Function to invoke 'plugin_write' function in python plugin
+ *
+ * @param    handle		Plugin handle from plugin_init_fn
+ * @param    name		Name of parameter to write
+ * @param    value		Value to be written to that parameter
+ */
+bool plugin_write_fn(PLUGIN_HANDLE handle, const std::string& name, const std::string& value)
+{
+	bool rv = false;
+
+	if (!handle)
+	{
+		Logger::getLogger()->fatal("plugin_handle: plugin_write(): "
+					   "handle is NULL");
+		return rv;
+	}
+
+	if (!pythonHandles)
+	{
+		Logger::getLogger()->error("pythonHandles map is NULL "
+					   "in plugin_write, plugin handle '%p'",
+					   handle);
+		return rv;
+	}
+
+	// Look for Python module for handle key
+	auto it = pythonHandles->find(handle);
+	if (it == pythonHandles->end() ||
+		!it->second ||
+		!it->second->m_module)
+	{
+		Logger::getLogger()->fatal("plugin_handle: plugin_write(): "
+					   "pModule is NULL, plugin handle '%p'",
+					   handle);
+		return rv;
+	}
+
+	std::mutex mtx;
+	PyObject* pFunc;
+	lock_guard<mutex> guard(mtx);
+	PyGILState_STATE state = PyGILState_Ensure();
+
+	Logger::getLogger()->debug("plugin_handle: plugin_write(): "
+				   "pModule=%p, handle=%p, plugin '%s'",
+				   it->second->m_module,
+				   handle,
+				   it->second->m_name.c_str());
+
+	// Fetch required method in loaded object
+	pFunc = PyObject_GetAttrString(it->second->m_module, "plugin_write");
+	if (!pFunc)
+	{
+		Logger::getLogger()->fatal("Cannot find method 'plugin_write' "
+					   "in loaded python module '%s'",
+					   it->second->m_name.c_str());
+	}
+
+	if (!pFunc || !PyCallable_Check(pFunc))
```

Comment:
If you return above you avoid getting the second long message the plunk is NULL. You will have to del with the GIL release of course.